### PR TITLE
Explicitly add DLL directories for Windows before importing

### PR DESCRIPTION
**Pull request by jacobperron**
New in Python 3.8, we should call os.add_dll_directory for directories
containing the DLLs we intend to import as well as their recursive
dependencies.

Connects to https://github.com/ros2/ci/pull/432

### CODE REVIEW

File: rosbag2_transport/rosbag2_transport/__init__.py [hunk 1]
```diff
@@ -28,5 +33,4 @@ def _import(name):
                          '#import-failing-even-with-library-present-on-the-system')
         raise
 
-
```
Comment (dirk-thomas) [inline rosbag2_transport/rosbag2_transport/__init__.py:31]: This seems unrelated and the linter might complain.

Comment (jacobperron) [reply rosbag2_transport/rosbag2_transport/__init__.py]: Fixed 3447ca4

**Comment by jacobperron** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/10797/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/6182/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/8790/)
* Windows [Build Status](https://ci.ros2.org/job/ci_windows/10691/)
